pkg/e2e/verify/nodepool.go

```diff
@@ -0,0 +1,474 @@
+			namePrefix: "resource-intensive-test-",
+			nodeName:   targetNode.Name,
+			image:      "registry.access.redhat.com/ubi8/ubi-minimal:latest",
+			command:    []string{"/bin/sh", "-c", "echo 'Resource test' && sleep 10"},
+			cpuRequest: "500m",
+			memRequest: "512Mi",
+			cpuLimit:   "1000m",
+			memLimit:   "1Gi",
+		})
+
+		err = client.Create(ctx, pod)
+		expect.NoError(err, "Failed to create resource-intensive pod")
+		defer client.Delete(ctx, pod)
+
+		err = wait.For(conditions.New(client).PodPhaseMatch(pod, corev1.PodSucceeded),
+			wait.WithTimeout(5*time.Minute), wait.WithInterval(5*time.Second))
+		expect.NoError(err, "Resource-intensive workload failed")
+	})
+
+	ginkgo.It("rejects operations on non-existent NodePools", func(ctx context.Context) {
+		if !canAccessAPI {
+			ginkgo.Skip("NodePool API not accessible")
+		}
+
+		_, err := h.Dynamic().Resource(nodePoolGVR).Namespace(clusterNamespace).
+			Get(ctx, "non-existent-nodepool", metav1.GetOptions{})
+
+		Expect(err).To(HaveOccurred(), "Should fail accessing non-existent NodePool")
+		Expect(apierrors.IsNotFound(err)).To(BeTrue(), "Should return NotFound error")
+	})
+
+	ginkgo.It("rejects duplicate NodePool creation", func(ctx context.Context) {
+		if !canAccessAPI {
+			ginkgo.Skip("NodePool API not accessible")
+		}
+
+		if !createdNodePool {
+			ginkgo.Skip("No NodePool created to test duplication")
+		}
+
+		duplicateSpec := buildNodePoolSpec(testNodePoolName, clusterNamespace)
+		duplicateObj := &unstructured.Unstructured{Object: duplicateSpec}
+
+		_, err := h.Dynamic().Resource(nodePoolGVR).Namespace(clusterNamespace).
+			Create(ctx, duplicateObj, metav1.CreateOptions{})
+
+		Expect(err).To(HaveOccurred(), "Should fail creating duplicate NodePool")
+		Expect(apierrors.IsAlreadyExists(err)).To(BeTrue(), "Should return AlreadyExists error")
+	})
+})
+
+type podConfig struct {
+	namePrefix   string
+	nodeName     string
+	nodeSelector map[string]string
+	image        string
+	command      []string
+	cpuRequest   string
+	memRequest   string
+	cpuLimit     string
+	memLimit     string
+}
+
+func createPod(namespace string, config *podConfig) *corev1.Pod {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: config.namePrefix,
+			Namespace:    namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{{
+				Name:    "test",
+				Image:   config.image,
+				Command: config.command,
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse(config.cpuRequest),
+						corev1.ResourceMemory: resource.MustParse(config.memRequest),
+					},
+				},
+			}},
+			RestartPolicy: corev1.RestartPolicyNever,
+		},
+	}
+
+	if config.nodeName != "" {
+		pod.Spec.NodeName = config.nodeName
+	}
+
+	if config.nodeSelector != nil {
+		pod.Spec.NodeSelector = config.nodeSelector
+	}
+
+	if config.cpuLimit != "" && config.memLimit != "" {
```

Comment:
Do you really need resource limits?  Aren't these short-lived pods?  Are you anticipating resource spikes that would need to be throttled?